app/controllers/MakePaymentController.scala

```diff
@@ -78,6 +70,8 @@ class MakePaymentController @Inject()(val messagesApi: MessagesApi,
               routes.MakePaymentController.makePayment().url
             )
             Redirect(url)
+          }.recover {
```

Comment:
Don't code by exception. If the service receives an error from the connector the service should return a `Left(error)` and the controller should return the appropriate view and status to the user.